embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/testcases/SpanTest.kt

```diff
@@ -47,17 +49,45 @@ internal class SpanTest {
 
 internal class FakeSpanExporter : SpanExporter {
     val exportedSpans = mutableListOf<SpanData>()
+    private val latches = mutableListOf<CountDownLatch>()
 
     override fun export(spans: MutableCollection<SpanData>): CompletableResultCode {
-        exportedSpans.addAll(spans)
+        synchronized(exportedSpans) {
+            exportedSpans.addAll(spans)
+            latches.forEach { latch ->
+                repeat(min(latch.count.toInt(), spans.size)) {
+                    latch.countDown()
+                }
+            }
+            latches.removeIf { it.count == 0L }
+        }
         return CompletableResultCode.ofSuccess()
     }
 
     override fun flush(): CompletableResultCode {
+        synchronized(exportedSpans) {
+            exportedSpans.clear()
+        }
         return CompletableResultCode.ofSuccess()
     }
 
     override fun shutdown(): CompletableResultCode {
         return CompletableResultCode.ofSuccess()
     }
+
+    /**
+     * Block the thread until the number of spans expected have been exported by this exporter since the last flush
```

Comment:
Theres's a small bug in this given how this function's behaviour is defined: when the exporter is flushed, the latch countdown values are persisted, but the comparison is for the number of exported spans AFTER the flush. 

So basically the latches take into account flushed spans, while invocations that happen after won't.

This doesn't affect this test, but the inconsistency is bad, so I'll be fixing this in the next PR to always look at all spans exported instead even before the flush.

Reply:
Never mind - I'm merging the two fixes into one PR